### PR TITLE
Update UI with Split Interface

### PR DESCRIPTION
**Pull request by wesho1107**
Update UI for application to support tab panes, new features include:
- TabPane UI
- Tab Commands
- ViewAssignments display on new pane

### CODE REVIEW

File: src/main/resources/view/MainWindow.fxml [hunk 1]
```diff
@@ -3,20 +3,24 @@
...
+<?import javafx.scene.layout.HBox?>
 <?import javafx.scene.layout.StackPane?>
 <?import javafx.scene.layout.VBox?>
+<?import javafx.stage.Stage?>
 
-<?import javafx.scene.layout.HBox?>
-<?import javafx.scene.control.Label?>
-<fx:root type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/17" xmlns:fx="http://javafx.com/fxml/1"
-         title="F.A.K.E.J.A.R.V.I.S." minWidth="600" minHeight="600" onCloseRequest="#handleExit">
+<fx:root minHeight="600" minWidth="600" onCloseRequest="#handleExit" title="F.A.K.E.J.A.R.V.I.S." type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/20.0.1" xmlns:fx="http://javafx.com/fxml/1">
```
Comment (raydenlim) [inline src/main/resources/view/MainWindow.fxml]: might want to split this into 2 lines, its more than 120 char (i think)

File: src/main/resources/view/MainWindow.fxml [hunk 2]
```diff
@@ -34,48 +38,75 @@
...
+                          <Tab text="Tasks">
+                              <VBox fx:id="taskList" VBox.vgrow="ALWAYS">
+                                  <Label maxWidth="Infinity" styleClass="section-label" text="Task List" wrapText="true" />
+                                  <StackPane fx:id="taskListPanelPlaceholder" VBox.vgrow="ALWAYS" />
+                              </VBox>
+                          </Tab>
+
+                          <Tab text="Assignments">
+                              <VBox fx:id="assignmentList" VBox.vgrow="ALWAYS">
+                                  <Label fx:id="assignmentListLabel" maxWidth="Infinity" styleClass="section-label" text="\$assignmentListLabel" wrapText="true" />
```
Comment (raydenlim) [inline src/main/resources/view/MainWindow.fxml]: This one is 129 char

File: src/main/resources/view/MainWindow.fxml (hunk 1 above)
Comment (raydenlim) [reply src/main/resources/view/MainWindow.fxml]: this line is 200 char long

**Review by raydenlim — COMMENTED** [review]
No major errors, LGTM!

File: src/main/resources/view/MainWindow.fxml [hunk 3]
```diff
@@ -34,48 +30,97 @@
...
-        <StackPane fx:id="statusbarPlaceholder" VBox.vgrow="NEVER" />
+          <HBox styleClass="pane-with-border" VBox.vgrow="ALWAYS">
+              <children>
+                  <TabPane maxHeight="Infinity" prefWidth="460" tabClosingPolicy="UNAVAILABLE" VBox.vgrow="ALWAYS">
+                      <tabs>
+                          <Tab text="Students">
+                              <VBox fx:id="personList" VBox.vgrow="ALWAYS">
+                                  <Label maxWidth="Infinity"
+                                         styleClass="section-label"
+                                         text="Person List"
```
Comment (raydenlim) [inline src/main/resources/view/MainWindow.fxml]: Student List :D

**Review by raydenlim — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 4]
```diff
@@ -198,6 +191,21 @@ public void handleHelp() {
         }
     }
 
+    /**
+     * Switches to Assignments tab.
+     */
+    public void handleViewAllAssignments() {
+        selectTab(3);
```
Comment (raydenlim) [inline src/main/java/seedu/address/ui/MainWindow.java]: magic number 3 :(

File: src/main/java/seedu/address/logic/commands/CommandType.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package seedu.address.logic.commands;
+/**
+ * Enum representing the type of commands.
+ */
+public enum CommandType {
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/commands/CommandType.java:5]: hmm, might be better to separate the words with underscore? similar to constants.
(convention for constants)
e.g ADD_TASK

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 6]
```diff
@@ -229,28 +246,49 @@ private CommandResult executeCommand(String commandText) throws CommandException
...
                 handleExit();
+                break;
+            case VIEWALLASSIGNMENTS:
+                handleViewAllAssignments();
+                break;
+            case VIEWASSIGNMENTS:
+                handleViewAssignments();
+                break;
+            case ADD:
+            case ADDTASK:
```
Comment (raydenlim) [inline src/main/java/seedu/address/ui/MainWindow.java]: Might have to SLAP this, code >30 LOC

**Review by raydenlim — COMMENTED** [review]
minor issues, but other than that, LGTM

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 4 above)
Comment (wesho1107) [reply src/main/java/seedu/address/ui/MainWindow.java]: WHOOPS

File: src/main/java/seedu/address/logic/commands/CommandType.java (hunk 5 above)
Comment (wesho1107) [reply src/main/java/seedu/address/logic/commands/CommandType.java:5]: Alright, will do

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 6 above)
Comment (wesho1107) [reply src/main/java/seedu/address/ui/MainWindow.java]: I believe enum is a special case?

File: src/main/java/seedu/address/logic/Messages.java [hunk 7]
```diff
@@ -31,6 +31,7 @@ public class Messages {
                 "Multiple values specified for the following single-valued field(s): ";
     public static final String MESSAGE_ASSIGNMENT_GRADED = "The assignment has already been graded";
     public static final String MESSAGE_ASSIGNMENT_UNGRADED = "The assignment has not been graded";
+    public static final String MESSAGE_INVALID_TAB_INDEX = "The inputted index is out of valid range";
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/Messages.java:34]: I've nvr seen the word `inputted` before. Did a search, and wow there are 2 past tenses for input.
input and inputted are accepted.

File: src/main/java/seedu/address/logic/commands/CommandType.java (hunk 5 above)
Comment (raydenlim) [reply src/main/java/seedu/address/logic/commands/CommandType.java:5]: Good Job bro

File: src/main/java/seedu/address/logic/commands/TabCommand.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Switches tab based on index.\n"
+            + "Parameters: INDEX (ranging from 1 for Students to 5 for Consultations\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_SUCCESS = "Switched to tab %1$s";
+    public static final int VALID_TAB_INDEX_LOWER_BOUND = 1;
+    public static final int VALID_TAB_INDEX_UPPER_BOUND = 5;
+
+    public static final CommandType COMMAND_TYPE = CommandType.TAB;
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/commands/TabCommand.java:24]: Codecov angry

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 9]
```diff
@@ -166,6 +167,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case ViewAllAssignmentsCommand.COMMAND_WORD:
             return new ViewAllAssignmentsCommandParser().parse(arguments);
 
+        case TabCommand.COMMAND_WORD:
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:170]: umbrella sad

File: src/main/java/seedu/address/logic/parser/TabCommandParser.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.TabCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new TabCommand object
+ */
+public class TabCommandParser implements Parser<TabCommand> {
```
Comment (raydenlim) [inline src/main/java/seedu/address/logic/parser/TabCommandParser.java:11]: cov

**Review by raydenlim — APPROVED** [review]
LGTM!

**Pull request merged**